### ISSUE TITLE
User story - Job Post Update

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#30]
### If applicable, add a related Epic.

Epic: #28 

### User Story

As a **recruiter user**, I can **update the details of my job post** so that I can **ensure the most up-to-date information about the post is provided**.

### Acceptance Criteria

- When logged in, the recruiter user should be able to edit their job postings
- The UI to edit the post should be similar to the post creation UI

### Please ensure you've completed these tasks before marking the issue as done.

- [x] Reviewed the user story for clarity and completeness.
- [x] Identified and linked related issues or epics.
- [x] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [ ] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [x] Updated the issue with any relevant notes or feedback (to date).
- [x] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [x] Create the url for `/user/<post_id>/edit`
- [x] Make sure that the frontend form is validated correctly
- [x] Ensure the database info is updated correctly
